File: coding_interview_pattern/sliding_window/maxSubArraySum.py
def maxSubArraySum(arr, n, K):
    """Compute subarray sum of size K
    """
    wstart = 0
    subSum = 0
    max = 0

    for i in range(n):
        subSum += arr[i]
        if i >= K-1:
            if max < subSum:
                max = subSum
            subSum -= arr[wstart]
            wstart += 1
    return max

if __name__ == "__main__":
    arr = [1, 5, -1, 6, 3, 2]
    size = len(arr)
    K = 3

    max = maxSubArraySum(arr, size, K)
    print(max)

File: coding_interview_pattern/sliding_window/C/max_sub_array_sum.c
#include <stdio.h>

int maxSubarraySum(int arr[], int n, int k)
{
    int i;
    int wstart = 0;
    int subsum = 0;
    int max = 0;

    for (i = 0; i < n; i++)
    {
        subsum += arr[i];

        if (i >= k-1)
        {
            max = max > subsum? max : subsum;
            subsum -= arr[wstart];
            wstart++;
        }
    }
    return max;
}


int main(void)
{
    int arr[6] = {1, 5, -1, 6, 3, 2};
    int size = sizeof(arr) / sizeof(arr[0]);
    int K = 3;

    int max = maxSubarraySum(arr, size, K);
    printf("%d\n", max);

    return (0);
}
